src/query/api/v1/handler/prometheus/common.go

```diff
@@ -417,22 +419,191 @@ func RenderSeriesMatchResultsJSON(
 	return jw.Close()
 }
 
-// PromResp represents Prometheus's query response.
-type PromResp struct {
+// Response represents Prometheus's query response.
+type Response struct {
+	// Status is the response status.
 	Status string `json:"status"`
-	Data   struct {
-		ResultType string `json:"resultType"`
-		Result     []struct {
-			Metric map[string]string `json:"metric"`
-			// todo(braskin): use `Datapoints` instead of interface{} in values
-			// Values is [float, string]
-			Values [][]interface{} `json:"values"`
-		} `json:"result"`
-	} `json:"data"`
+	// Data is the response data.
+	Data data `json:"data"`
+}
+
+type data struct {
+	// ResultType is the result type for the response.
+	ResultType string `json:"resultType"`
+	// Result is the list of results for the response.
+	Result results `json:"result"`
+}
+
+type results []Result
+
+// Len is the number of elements in the collection.
+func (r results) Len() int { return len(r) }
+
+// Less reports whether the element with
+// index i should sort before the element with index j.
+func (r results) Less(i, j int) bool {
+	return r[i].id < r[j].id
+}
+
+// Swap swaps the elements with indexes i and j.
+func (r results) Swap(i, j int) { r[i], r[j] = r[j], r[i] }
+
+func (r results) sort() {
+	for _, result := range r {
+		result.genID()
+	}
+
+	sort.Sort(r)
+}
+
+// Result is the result itself.
+type Result struct {
+	// Metric is the tags for the result.
+	Metric Tags `json:"metric"`
+	// Values is the set of values for the result.
+	Values Values `json:"values"`
+	id     string
+}
+
+// Tags is a simple representation of Prometheus tags.
+type Tags map[string]string
+
+// Values is a list of values for the Prometheus result.
+type Values []Value
+
+// Value is a single value for Prometheus result.
+type Value []interface{}
+
+func (r *Result) genID() {
+	var sb strings.Builder
+	// NB: this may clash but exact tag values are also checked, and this is a
+	// validation endpoint so there's less concern over correctness.
```

Comment:
Hm, might it problematic if someone pulls in this type for a real endpoint at some point (since it's in the `prometheus` package as common code)?

Should be as simple as:
```go
tags := models.NewTags(0, nil)
for k, v := range r.Metric {
  tags.Add(model.Tag{Name: []byte(k), Value: []byte(v)})
}
r.id = string(tags.ID())
```

Reply:
I'm honestly leaning more to the side of removing this, it's annoying to maintain and the comparator should cover all it was meant to do

Reply:
I think I may leave it now and excise it along with some other annoying quirks later on